tt_metal/impl/device/mesh_device.hpp

```diff
@@ -174,11 +174,11 @@ class MeshDevice : public std::enable_shared_from_this<MeshDevice> {
 
     static std::shared_ptr<MeshDevice> fetch_mesh_device(const std::vector<Device*>& devices);
     static std::shared_ptr<MeshDevice> create(
-        size_t l1_small_size,
-        size_t trace_region_size,
-        size_t num_command_queues,
-        DispatchCoreType dispatch_core_type,
-        const MeshDeviceConfig &config);
+        const MeshDeviceConfig &config,
+        size_t l1_small_size = DEFAULT_L1_SMALL_SIZE,
```

Comment:
I didn't want to block the PR given it is to restore pipeline health but why the move to so many default values? And can we avoid the repeated types with default values to make the API less error prone?

Reply:
I totally agree with you. This is a reflection of how our tt-metal Device constructor looks today. All of these arguments are used to pipe into `Device` constructor. So you'll see the identical constructor there. So while not ideal, I'm expecting the tt-metal API cleanup efforts to clean all this up up in one-shot. For this PR, it's outside the scope.

Reply:
See: https://github.com/tenstorrent/tt-metal/pull/12681/files#diff-76e63af42b6861192b5e9e0e16291987d68b7c99d7d739e4df9d08fbf012e7c1R63